matter_server/common/helpers/util.py

```diff
@@ -159,6 +160,10 @@ def parse_value(
                 return parse_value(name, value, sub_arg_type)
             except (KeyError, TypeError, ValueError):
                 pass
+        # Before we give up, check if we have a ValueDecodeFailure
+        if isinstance(value, dict) and "TLVValue" in value and "Reason" in value:
+            return ValueDecodeFailure(value["TLVValue"], value["Reason"])
```

Comment:
Where and how will this instance be handled downstream?

Reply:
Well currently it's not handled at all, it just gracefully replaces the correct field with that which is the same behavior the chip cli has, it could probably be set to some default to make sure it's the correct type instead of returning `ValueDecodeFailed`

Reply:
idk which approach you guys would prefer for this, I do think handling it gracefully is important as devices having unparsable values is a real possibility like the hue bridge but at the same time getting errors is also important so maybe adding a strict mode that errors out and a normal mode that just gracefully returns is a good solution?

Reply:
I think we need to think about and discuss how we want to handle this where we use the util, both here in this library and when library users use the util, eg Home Assistant.

What are the different usages and what should happen when we encounter this case?

Reply:
Well the only usage I've seen for this is to get specific attributes from a cluster without having to subscribe to events, some examples would be getting the min/max brightness of a light, getting the min/max values for color, and in my specific case, it was checking if the cluster has a specific feature enabled using the featureMap attribute, so it's mostly about getting static data from a device.

As to handling this from what I've seen on the only other matter controller I have working which is apple home they just ignore it completely as the device works, now this might be because looking at the matter spec this is only needed if a specific feature is enabled in this case color loop which hue has disabled, with that being said they should still be returning parsable data for this attribute, in this case, it should be a null. So I think we should just return a `ValueDecodeFailure` if that's what we get from the matter API and we should also add a function to get specific attributes from a cluster which should error out if the value is of an incorrect type, something that should be simple enough to add since the `MatterNode` class already has a function for fetching specific attributes it just doesn't check if the type is correct.